### PR TITLE
upgrade to work with esri js api 4.x

### PR DESCRIPTION
**Pull request by stdavis**
image

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -6,7 +6,9 @@
         "node": true
     },
     "globals": {},
+    "plugins": ["html"],
     "rules": {
+        "semi": 2,
```
Comment (steveoh) [inline .eslintrc]: what does 2 mean? I like the style from atlas as it's more readable.

File: Sherlock.js [hunk 2]
```diff
@@ -191,55 +187,39 @@ define([
...
-                this.symbolFill = new SimpleFillSymbol(
-                    SimpleFillSymbol.STYLE_NULL,
-                    new SimpleLineSymbol(
-                        SimpleLineSymbol.STYLE_DASHDOT,
-                        new Color([255, 255, 0]),
-                        1.5),
-                    null);
+                this.symbolFill = new SimpleFillSymbol({
+                    style: 'none',
+                    outline: {
```
Comment (steveoh) [inline Sherlock.js:159]: gosh this is much nicer.

File: Sherlock.js [hunk 3]
```diff
@@ -548,30 +528,21 @@ define([
...
-                        // swallow errors from cancels
-                        if (err.message !== 'undefined') {
-                            throw new Error('sherlock.Sherlock Provider Error: ' + err.message);
-                        }
 
-                        this.hideSpinner();
-                    }
-                ));
+                    this.hideSpinner();
+                }.bind(this)
```
Comment (steveoh) [inline Sherlock.js]: did you want to go full bind and remove all hitch?

File: Sherlock.js [hunk 4]
```diff
@@ -590,41 +561,52 @@ define([
...
-            if (graphic.geometry.type === 'point') {
-                // zoom and center on point
-
-                // get base layer
-                var bLayer = this.map.getLayer(this.map.layerIds[0]);
-
-                // get next to lowest lod
-                var lod = bLayer.tileInfo.lods.length - this.zoomLevel;
+            if (graphics[0].geometry.type === 'point') {
+                var lod = this.mapView.map.basemap.baseLayers.items[0].tileInfo.lods.length - this.zoomLevel;
```
Comment (steveoh) [inline Sherlock.js]: this seems goofy. is there not a more straight forward way?

File: .eslintrc (hunk 1 above)
Comment (stdavis) [reply .eslintrc]: Agreed. Fixed with switch to `eslint-config-agrc`.

File: Sherlock.js (hunk 2 above)
Comment (stdavis) [reply Sherlock.js:159]: Oh ya.

File: Sherlock.js (hunk 3 above)
Comment (stdavis) [reply Sherlock.js]: Removed all hitches during babelify commit.

File: Sherlock.js (hunk 4 above)
Comment (stdavis) [reply Sherlock.js]: Not that I could find.

**Comment by stdavis** [comment]
OK @steveoh. This is ready for another look.

image

**Review by steveoh — DISMISSED** [review]
this all seems good and dandy to me.

**Comment by steveoh** [comment]
I suppose my only concern is checking source maps into the main repo. Don't you think that is more of a dev concern? If they are ignored by bower or npm they will still be requested because of the markup in the files.

**Comment by stdavis** [comment]
I think that it's handy to have them when you are debugging an app wondering if it's your code or a dependency.

**Pull request merged**